app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -421,14 +421,31 @@ export default Controller.extend(bufferedProperty("model"), {
       }
     },
 
-    toggleSummary() {
+    showSummary() {
       return this.get("model.postStream")
-        .toggleSummary()
+        .showSummary()
         .then(() => {
           this.updateQueryParams();
         });
     },
 
+    cancelFilter(previousFilters) {
+      this.get("model.postStream").cancelFilter();
+      this.get("model.postStream")
+        .refresh()
+        .then(() => {
+          if (previousFilters) {
+            if (previousFilters.replies_to_post_number) {
+              this._jumpToPostNumber(previousFilters.replies_to_post_number);
```

Comment:
can we use DiscourseURL.jumpToPost here (and after), didn't check the actual implementation of _jumpToPostNumber

Reply:
I would prefer to keep `_jumpToPostNumber` and `_jumpToPostID`, under the hood they both use `DiscourseURL.routeTo`. They are similar to DiscourseURL.jumpToPost, except that the latter only works for post numbers and we would need to redo the same thing as `_jumpToPostID` to get it to go to a post by ID.